examples/ex_adaptive.ipynb

```diff
@@ -0,0 +1,503 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def relative_diff(x: float, y: float) -> float:\n",
+    "    \"\"\"Relative difference\"\"\"\n",
+    "    if max(abs(x), abs(y)) == 0:\n",
+    "        return 0\n",
+    "    return abs(x-y)/max(abs(x), abs(y))\n",
+    "relative_diff = np.vectorize(relative_diff)\n",
+    "\n",
+    "def absolute_diff(x: ArrayLike, y: ArrayLike):\n",
+    "    \"\"\"Absolute difference\"\"\"\n",
+    "    return np.abs(x-y)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Define a function\n",
+    "\n",
+    "Here, we choose a function where we know analytically what the Fourier integral is, for testing purposes. Of course, in a real setting, one would choose a function without any analytically known Fourier integral."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def inv_sqrt(f: ArrayLike) -> ArrayLike:\n",
+    "    return 1 / np.sqrt(2*np.pi*f + 1e-50)\n",
+    "\n",
+    "# The analytic fourier integral of inv_sqrt\n",
+    "def inv_sqrt_analytic_integral(t: ArrayLike):\n",
+    "    return np.sqrt(np.pi / (2 * t))"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Define frequencies and times\n",
+    "Define a sorted array of frequencies as a starting point for the algorithm. It is possible to do as is done here and only provide two frequencies, and still get decent results. It is however *recommended* to follow the following two guidelines when giving an initial frequency array:\n",
+    "1. **Make sure all important features of the function are within the endpoints of your array**. The algorithm does a simple scan to higher and lower frequencies, but it is better at adding helpful points to the interior of the initial range.\n",
+    "2. **Try to supply a good initial guess of frequencies** that capture the important features of the function. The algorithm starts with doing a rough scan to search for features, then gradually hones in on a finer mesh around the features it finds. However, if there is too much space between the points in the starting array, the rough scan might miss important features.\n",
+    "\n",
+    "Examples of the effect of these guidelines will be showed below.\n",
+    "\n",
+    "It is possible to include `-np.inf`, `0` and/or `np.inf` in the frequency range. In the case of infinities, the algorithm will add progressively lower/higher frequencies until the tolerance is reached. When using infinities, some discretion is recommended, as for some functions and parameters, the algorithm may prioritize going towards the infinity at the expense of refining the interior of the range of finite frequencies.\n",
+    "\n",
+    "Also define the times you want to calculate the integral for."
```

Comment:
Maybe one could state that the times can be anything (they are not constrained, as in a std DFT). A similar statement could be made in the other example.